test/integ/k8s/resource.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package k8s
+
+import (
+	"context"
+	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+// DoesCRDExist returns true if the given CRD exists
+func (c Client) DoesCRDExist(crdName string) bool {
+	crds, err := c.apixClient.CustomResourceDefinitions().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		ginkgo.Fail("ginkgo.Failed to get list of CustomResourceDefinitions")
+	}
+	for i := range crds.Items {
+		if strings.Compare(crds.Items[i].ObjectMeta.Name, crdName) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// DoesClusterRoleExist returns true if the given ClusterRole exists
+func (c Client) DoesClusterRoleExist(name string) bool {
+	_, err := c.clientset.RbacV1().ClusterRoles().Get(context.TODO(), name, metav1.GetOptions{})
+	return procExistsStatus(err, "ClusterRole")
+}
+
+// DoesClusterRoleBindingExist returns true if the given ClusterRoleBinding exists
+func (c Client) DoesClusterRoleBindingExist(name string) bool {
+	_, err := c.clientset.RbacV1().ClusterRoleBindings().Get(context.TODO(), name, metav1.GetOptions{})
+	return procExistsStatus(err, "ClusterRoleBinding")
+}
+
+// DoesNamespaceExist returns true if the given Namespace exists
+func (c Client) DoesNamespaceExist(name string) bool {
+	_, err := c.clientset.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
+	return procExistsStatus(err, "Namespaces")
```

Comment:
Nit. should these names passed to procExistsStatus be plural?

Reply:
I can make it singular